src/main/java/io/fabric8/maven/docker/assembly/DockerAssemblyManager.java

```diff
@@ -203,32 +201,47 @@ private void interpolateDockerfile(File dockerFile, BuildDirs params, FixedStrin
 
     // visible for testing
     void verifyGivenDockerfile(File dockerFile, BuildImageConfiguration buildConfig, FixedStringSearchInterpolator interpolator, Logger log) throws IOException {
-        AssemblyConfiguration assemblyConfig = buildConfig.getAssemblyConfiguration();
-        if (assemblyConfig == null) {
+        List<AssemblyConfiguration> assemblyConfigs = buildConfig.getAssemblyConfigurations();
+        if (assemblyConfigs.isEmpty()) {
             return;
         }
 
+        for (AssemblyConfiguration assemblyConfig : assemblyConfigs) {
+            verifyAssemblyReferenced(dockerFile, interpolator, log, assemblyConfig);
+        }
+    }
+
+    private String firstNonOptionArgument(String[] line) {
+        int i;
+        for (i = 1; i < line.length; i++) {
+            String component = line[i];
+            if (!component.startsWith("--")) {
+                return line[i];
+            }
+        }
+
+        return null;
```

Comment:
nit, I think this can be simplified like this:
```java
return Arrays.stream(line)
        .skip(1)
        .filter(component -> !component.startsWith("--"))
        .findFirst()
        .orElse(null);
```

Reply:
That's much nicer; I'll do that instead.